src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationKeyVaultOptions.cs

```diff
@@ -34,5 +37,20 @@ public AzureAppConfigurationKeyVaultOptions Register(SecretClient secretClient)
             SecretClients.Add(secretClient);
             return this;
         }
+
+        /// <summary>
+        /// Sets the callback used to resolve key vault references that have no registered <see cref="SecretClient"/>.
+        /// </summary>
+        /// <param name="secretResolver">A callback that maps the <see cref="Uri"/> of the key vault secret to its value.</param>
+        public AzureAppConfigurationKeyVaultOptions SetSecretResolver(Func<Uri, ValueTask<string>> secretResolver)
```

Comment:
`ValueTask` indicates that we expect implementations to typically return in an a synchronous manner. Is this on purpose? I would expect secret resolution to be mostly an async operation.

Reply:
We should allow both sync and async operations because we don't want to assume how users resolve their secrets.

Reply:
Resolved this with Drago. `ValueTask` gives the chance for developers to be efficient when implementing an interface via synchronous completion. Whereas, if we went with `Task`, there wouldn't be any benefit over `ValueTask`. We do not rely on `Task` anywere (no `ContinueWith`) so `ValueTask` is a good option for us here.

Reply:
Leaving Stephen Toub's comment regarding `ValueTask` here as well for future reference.
https://github.com/dotnet/runtime/issues/25182#issuecomment-370609431